base/intset.jl

```diff
@@ -1,339 +1,232 @@
-symdiff(s1::IntSet, s2::IntSet) =
-    (s1.limit >= s2.limit ? symdiff!(copy(s1), s2) : symdiff!(copy(s2), s1))
-
-function empty!(s::IntSet)
-    fill!(s.bits, 0)
-    return s
+    s
 end
-
+intersect(s1::IntSet, s2::IntSet) =
+    (length(s1.bits) >= length(s2.bits) ? intersect!(copy(s1), s2) : intersect!(copy(s2), s1))
 """
-    symdiff!(s, n)
+    intersect!(s1::IntSet, s2::IntSet)
 
-The set `s` is destructively modified to toggle the inclusion of integer `n`.
+Intersects sets `s1` and `s2` and overwrites the set `s1` with the result. If needed, `s1`
+will be expanded to the size of `s2`.
 """
-function symdiff!(s::IntSet, n::Integer)
-    if n >= s.limit
-        lim = Int(n + div(n,2))
-        sizehint!(s, lim)
-    elseif n < 0
-        throw(ArgumentError("IntSet elements cannot be negative"))
-    end
-    s.bits[n>>5 + 1] ⊻= UInt32(1)<<(n&31)
-    return s
+function intersect!(s1::IntSet, s2::IntSet)
+    _matched_map!(&, s1.bits, s2.bits)
+    s1
+end
+
+setdiff(s::IntSet, ns) = setdiff!(copy(s), ns)
+setdiff!(s::IntSet, ns) = (for n in ns; _delete!(s, n); end; s)
+function setdiff!(s1::IntSet, s2::IntSet)
+    _matched_map!(>, s1.bits, s2.bits)
+    s1
 end
 
+symdiff(s::IntSet, ns) = symdiff!(copy(s), ns)
 """
     symdiff!(s, itr)
 
 For each element in `itr`, destructively toggle its inclusion in set `s`.
 """
-function symdiff!(s::IntSet, ns)
-   for n in ns
-       symdiff!(s, n)
-   end
-   return s
-end
+symdiff!(s::IntSet, ns) = (for n in ns; symdiff!(s, n); end; s)
+"""
+    symdiff!(s, n)
 
-function copy!(to::IntSet, from::IntSet)
-    if to === from
-        return to
-    else
-        empty!(to)
-        return union!(to, from)
-    end
+The set `s` is destructively modified to toggle the inclusion of integer `n`.
+"""
+function symdiff!(s::IntSet, n::Integer)
+    0 <= n < typemax(Int) || _throw_intset_bounds_err()
+    val = !(n in s)
+    _setint!(s, n, val)
+    s
 end
-
-in(n, s::IntSet) = n < 0 ? false : (n > typemax(Int) ? s.fill1s : in(convert(Int, n), s))
-function in(n::Int, s::IntSet)
-    if n >= s.limit
-        # max IntSet length is typemax(Int), so highest possible element is
-        # typemax(Int)-1
-        return s.fill1s && n >= 0 && n < typemax(Int)
-    elseif n <= 0
-        if n < 0
-            return false
-        else
-            depwarn("stored zeros in IntSet is deprecated", :in)
-        end
-    end
-    (s.bits[n>>5 + 1] & (UInt32(1)<<(n&31))) != 0
+function symdiff!(s1::IntSet, s2::IntSet)
+    _matched_map!(xor, s1.bits, s2.bits)
+    s1
 end
 
-start(s::IntSet) = Int64(0)
-done(s::IntSet, i) = (!s.fill1s && next(s,i)[1] >= s.limit) || i == typemax(Int)
-function next(s::IntSet, i)
-    if i >= s.limit
-        n = Int64(i)
+@inline function in(n::Integer, s::IntSet)
+    if 1 <= n <= length(s.bits)
+        @inbounds b = s.bits[n]
     else
-        n = Int64(ccall(:bitvector_next, UInt64, (Ptr{UInt32}, UInt64, UInt64), s.bits, i, s.limit))
```

Comment:
Some of the functions are used by flisp, but most aren't. I think everything after bitvector_get in bitvector.c can be deleted.